src/core/xml.cpp

```diff
@@ -433,14 +433,19 @@ static std::pair<std::string, std::string> parse_xml(XMLSource &src, XMLParseCon
         if (!param.empty()) {
             std::sort(param.begin(), param.end(),
                 [](const auto &a, const auto &b) -> bool {
-                    return a.first.length() > b.first.length();
+                    return std::get<0>(a).length() > std::get<0>(b).length();
                 });
+
             for (auto attr: node.attributes()) {
                 std::string value = attr.value();
                 if (value.find('$') == std::string::npos)
                     continue;
-                for (const auto &kv : param)
-                    string::replace_inplace(value, "$" + kv.first, kv.second);
+                for (auto &p : param) {
+                    if (string::replace_inplace(value, "$" + std::get<0>(p), std::get<1>(p))) {
+                        std::get<2>(p) = true;
+                        break;
```

Comment:
I would remove this `break` statement. With it, users wouldn't be able to have something like:
```
 <rgb name="radiance" value="$r_rad, $g_rad, $b_rad"/>
```
It's a weird use case, I admit.

Reply:
If we allow this, we need to make sure that variable names are well validated (e.g. no comma in them).

Reply:
In that case, even if we don't allow it we should have variable name checking, no?
```
<rgb name="radiance" value="$r_rad, 6.5, 2.3"/>
```
This example can still break.